apps/meteor/app/cloud/server/functions/supportedVersionsToken/supportedVersionsToken.ts

```diff
@@ -163,14 +162,28 @@ const getSupportedVersionsToken = async () => {
 				msg: 'Using supported versions from license',
 			});
 			break;
-		case response.success && response.result:
+		case cloudResponse.success && cloudResponse.result:
 			SystemLogger.info({
 				msg: 'Using supported versions from cloud',
 			});
 			break;
 	}
 
-	await buildVersionUpdateMessage(supportedVersions?.versions);
+	// to avoid a possibly wrong message, we only send the message if the cloud response was successful
+	if (cloudResponse.success) {
+		await buildVersionUpdateMessage(supportedVersions?.versions);
+	} else if (retry < 5) {
+		// in case of failure we'll try again later
+		setTimeout(
+			async () => {
+				await getCachedSupportedVersionsToken.reset(retry + 1);
```

Comment:
Should we retry for every failed response? I assume there are a few error codes where it would not make sense to retry

Reply:
Oh yea youre right my bad